js_modules/dagit/packages/core/src/workspace/asset-graph/AssetGraphExplorer.tsx

```diff
@@ -0,0 +1,278 @@
+import {gql, useQuery} from '@apollo/client';
+import {NonIdealState} from '@blueprintjs/core';
+import {IconNames} from '@blueprintjs/icons';
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {LATEST_MATERIALIZATION_METADATA_FRAGMENT} from '../../assets/LastMaterializationMetadata';
+import {SVGViewport} from '../../graph/SVGViewport';
+import {useDocumentTitle} from '../../hooks/useDocumentTitle';
+import {PipelineExplorerPath} from '../../pipelines/PipelinePathUtils';
+import {SidebarPipelineOrJobOverview} from '../../pipelines/SidebarPipelineOrJobOverview';
+import {PipelineExplorerSolidHandleFragment} from '../../pipelines/types/PipelineExplorerSolidHandleFragment';
+import {METADATA_ENTRY_FRAGMENT} from '../../runs/MetadataEntry';
+import {POLL_INTERVAL} from '../../runs/useCursorPaginatedQuery';
+import {ColorsWIP} from '../../ui/Colors';
+import {Loading} from '../../ui/Loading';
+import {SplitPanelContainer} from '../../ui/SplitPanelContainer';
+import {repoAddressToSelector} from '../repoAddressToSelector';
+import {RepoAddress} from '../types';
+
+import {AssetNode, getNodeDimensions} from './AssetNode';
+import {ForeignNode, getForeignNodeDimensions} from './ForeignNode';
+import {SidebarAssetInfo} from './SidebarAssetInfo';
+import {
+  buildGraphComputeStatuses,
+  buildGraphData,
+  buildSVGPath,
+  graphHasCycles,
+  layoutGraph,
+  Node,
+} from './Utils';
+import {
+  AssetGraphQuery,
+  AssetGraphQueryVariables,
+  AssetGraphQuery_repositoryOrError_Repository_assetNodes,
+} from './types/AssetGraphQuery';
+
+type AssetNode = AssetGraphQuery_repositoryOrError_Repository_assetNodes;
+
+interface Props {
+  repoAddress: RepoAddress;
+  explorerPath: PipelineExplorerPath;
+  handles: PipelineExplorerSolidHandleFragment[];
+  selectedHandle?: PipelineExplorerSolidHandleFragment;
+  onChangeExplorerPath: (path: PipelineExplorerPath, mode: 'replace' | 'push') => void;
+}
+
+export const AssetGraphExplorer: React.FC<Props> = (props) => {
+  const {repoAddress, handles, selectedHandle, explorerPath, onChangeExplorerPath} = props;
+  const repositorySelector = repoAddressToSelector(repoAddress);
+  const queryResult = useQuery<AssetGraphQuery, AssetGraphQueryVariables>(ASSETS_GRAPH_QUERY, {
+    variables: {repositorySelector},
+    notifyOnNetworkStatusChange: true,
+    pollInterval: POLL_INTERVAL,
```

Comment:
Is it important to poll this?

Reply:
Hmm good catch, I think we can rm this - it looks like the non-asset version of this explorer doesn't poll so this doesn't need to either.